utils/src/metrics.rs

```diff
@@ -0,0 +1,98 @@
+use prometheus_client::encoding::EncodeLabelSet;
+use prometheus_client::metrics::counter::Counter;
+use prometheus_client::metrics::family::Family;
+
+/// Metric label that indicates success or failure.
+#[derive(Clone, Debug, Hash, PartialEq, Eq, EncodeLabelSet)]
+pub struct Status {
+    pub ok: &'static str,
+}
+
+/// Success and failure labels.
+const SUCCESS: Status = Status { ok: "o" };
```

Comment:
To make it easier for downstream folks, I think it should be `status:"ok"/"fail"` rather than `"o"/"x"`.